tests/Doctrine/Tests/ORM/Mapping/XmlMappingDriverTest.php

```diff
@@ -149,27 +149,69 @@ public function testInvalidMappingFileException()
      * @param string $xmlMappingFile
      * @dataProvider dataValidSchema
      * @group DDC-2429
+     * @group 6389
      */
     public function testValidateXmlSchema($xmlMappingFile)
+    {
+        $this->assertTrue($this->doValidateXmlSchema($xmlMappingFile));
+    }
+
+    /**
+     * @param string $xmlMappingFile
+     * @dataProvider dataInvalidSchema
+     * @group 6389
+     */
+    public function testInvalidateXmlSchema($xmlMappingFile)
+    {
+        // Avoid PHP Warning errors for expected invalid mappings
```

Comment:
What errors did come up here?

Reply:
Asking because this part should probably be moved into the XML driver itself

Reply:
```
$ ./vendor/bin/phpunit --group 6389
PHPUnit 6.5.9 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.7-2+0~20180714182139.1+stretch~1.gbp3fcba8
Configuration: /z/phpunit.xml.dist

.................................................EEEEEEEEEEEEEE   63 / 63 (100%)

Time: 1.11 seconds, Memory: 58.00MB

There were 14 errors:

1) Doctrine\Tests\ORM\Mapping\XmlMappingDriverTest::testInvalidateXmlSchema with data set #0 ('/z/tests/Doctrine/Tests/ORM/M...cm.xml')
DOMDocument::schemaValidate(): Element '{http://doctrine-project.org/schemas/orm/doctrine-mapping}class': This element is not expected. Expected is one of ( {http://doctrine-project.org/schemas/orm/doctrine-mapping}mapped-superclass, {http://doctrine-project.org/schemas/orm/doctrine-mapping}entity, {http://doctrine-project.org/schemas/orm/doctrine-mapping}embeddable, ##other{http://doctrine-project.org/schemas/orm/doctrine-mapping}* ).

/z/tests/Doctrine/Tests/ORM/Mapping/XmlMappingDriverTest.php:185
/z/tests/Doctrine/Tests/ORM/Mapping/XmlMappingDriverTest.php:170

2) Doctrine\Tests\ORM\Mapping\XmlMappingDriverTest::testInvalidateXmlSchema with data set #1 ('/z/tests/Doctrine/Tests/ORM/M...20.xml')
DOMDocument::schemaValidate(): Element '{http://doctrine-project.org/schemas/orm/doctrine-mapping}custom-id-generator', attribute 'class': [facet 'pattern'] The value 'Foo Bar' is not accepted by the pattern '([a-zA-Z_-􏿿][a-zA-Z0-9_-􏿿]*\\)*[a-zA-Z_-􏿿][a-zA-Z0-9_-􏿿]*'.

/z/tests/Doctrine/Tests/ORM/Mapping/XmlMappingDriverTest.php:185
/z/tests/Doctrine/Tests/ORM/Mapping/XmlMappingDriverTest.php:170

*SNIP*

ERRORS!
Tests: 63, Assertions: 49, Errors: 14.
```

Alternatively it works with `@expectedException \PHPUnit\Framework\Error\Warning` (or do you prefer `$this->expectException(\PHPUnit\Framework\Error\Warning::class);`?) and reducing the body to the simple call
`$this->doValidateXmlSchema($xmlMappingFile);`
but that's depending both on `DOMDocument::schemaValidate()` emitting an `E_WARNING` and on PHPUnit translating the Warning to an exception...

This is only for testing XSD validation (actually testing that the schema definition is correct), how would you move it into the XML driver (without merging #6728)?

Reply:
@greg0ire ping ^

Reply:
@guilliamxavier You could reuse the `libxml_use_internal_errors` yadda yadda trick from https://github.com/doctrine/doctrine2/pull/6728/files#diff-f96b6283e2653400c7e5fbb845f50748R867 here.

@Ocramius not 100% sure why you ping me, if it is to know the status of #6728, it is blocked by #7199, which is blocked until further notice